allennlp/service/config_explorer.html

```diff
@@ -103,511 +190,558 @@ <h2>AllenNLP Configuration Wizard (alpha)</h2>
+                !(configurable || origin === 'Dict' || origin === 'List' || origin == 'Sequence' || (origin == 'Tuple' && args.size == 2 && args.get(2) === '...'))
+        }
+
+        // One of the key-value pairs that makes up a Config.
+        const ConfigItem = ({ path, item, setData }) => {
+            const name = item.get('name')
+            const annotation = item.get('annotation')
+            const configurable = item.get('configurable')
+            const defaultValue = item.get('defaultValue')
+            const comment = item.get('comment')
+            const completed = item.get('completed')
+
+            const classNames = [
+                'config-item',
+                defaultValue === undefined ? 'required' : 'optional',
+                completed ? 'complete' : 'incomplete'
+            ].join(' ')
+
+            const showRequired = defaultValue === undefined && !completed
+            const showInline = showItemInline(item, false)
+            const tooltip = comment ? <Tooltip title={comment} /> : null
+            const isRequired = showRequired ? <RequiredWarning title="Required" /> : null
+            const renderedAnnotation = <span class="annotation">({renderAnnotation(annotation)})</span>
+
+            return (
+                <li className={classNames}>
+                    <section>
+                        {isRequired}
+                        <span class="name">{name} {tooltip}</span>
+                        {renderedAnnotation}
+                        {showInline ? renderChild(path, item, setData) : null}
+                    </section>
+                    {!showInline ? <ul>{renderChild(path, item, setData)}</ul> : null}
+                </li>
+            )
+        }
+
+        // Find the longest common prefix
+        const sharedPrefix = (s1, s2) => {
+            for (let i = 0; i < s1.length; i++) {
+                if (s1[i] !== s2[i]) {
+                    return s1.slice(0, i)
                 }
+            }
+            // Otherwise, all of s1
+            return s1
+        }
+
+        const commonPrefix = (strings) => {
+            if (strings.size == 1) {
+                // Special case, return up to last .
+                const [string] = strings
+                const idx = string.lastIndexOf(".")
+                return string.slice(0, idx + 1)
+            } else {
+                return strings.reduce(sharedPrefix)
+            }
+        }
+
+        const Configurator = ({ path, item, setData }) => {
+            const annotation = item.get('annotation')
+            const className = annotation.get('origin')
+
+            // User-supplied
+            const choices = item.get('choices')
+            const choice = item.get('choice')
+            const config = item.get('value')
+
+            let prefix = choices ? commonPrefix(choices) : null
+            prefix = (prefix && prefix.length) ? prefix.substring(0, prefix.length - 1) : null // remove last dot
```

Comment:
I thought it looked funny....
instead of looking at it as a namespace.... it feel more like an option header...
bla.blas [option.a V]
vs
bla.blas. [option.a V]

IMO the change looks better... but you are the owner.. so let me know

Reply:
yeah, I prefer having the dot outside the dropdown, I think it looks odd inside

Reply:
i dont think i added it inside... just removed it from the label.... but ill put it back

Reply:
well, it used to be outside, and now it's inside, so you're the most likely culprit 😀

Reply:
yes, i did in fact do it ;-P
ill revert...
anything else to fix up??

Reply:
the intent of my change was to remove the dot form both places...
the effect of my change was to move the dot inside the dropdown
(neither of us prefer the do inside the dropdown)

i have reverted my change... but i still argue that no dot would be best....
cheers